app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -3928,4 +3928,63 @@ class PatientRepositoryAndroidTest {
         patientWithCurrentFacilityAsAssignedFacility, patientWithAnotherFacilityAsAssignedFacility
     )
   }
+
+  @Test
+  fun `searching_for_a_patient_by_name_must_return_list_of_patient_search_results_paritioned_by_assigned_and_not_assigned_facility`() {
+    fun createPatientWithNameAndAssignedFacilityID(
+        patientUuid: UUID,
+        patientName: String,
+        assignedFacilityId: UUID?
+    ) {
+      val patientProfile = testData
+          .patientProfile(patientUuid = patientUuid, patientName = patientName, patientAssignedFacilityId = assignedFacilityId)
+          .let { patientProfile ->
+            patientProfile.copy(
+                patient = patientProfile.patient.copy(fullName = patientName, assignedFacilityId = assignedFacilityId),
+            )
+          }
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+    }
+
+    fun searchResults(patientName: String, facilityUuid: UUID): Set<UUID> {
+      return patientRepository
+          .search2(Name(patientName), facilityUuid)
+          .map { it.uuid }
+          .toSet()
```

Comment:
Is there a reason we are converting them to `Set` here?

Reply:
Yes, because I am setting multiple values of search results.

Reply:
yeah, but wouldn't a list suffice for this?

Reply:
I mean, sure. Is there a reason we should not be using a set?

Reply:
When we are using `toSet` we are constructing a new `Set<T>` or ` Collection<T>`, while it's not that big of a problem here we don't have to do that if we don't need it right?

Reply:
But, while using a list we will be creating a new list as well right?

Reply:
Yes, but the `toSet` operation is done on that particular list. Here we have used the `map` operator to generate a list of specific fields, and we are applying `toSet` on top of that. Is there is no specific reason for using the `toSet` extension, we can use the list from `map` function. So here we can remove the `toSet` extension and it should still behave the same way.

Reply:
Got it